src/queries/label-text.js

```diff
@@ -45,73 +58,94 @@ function queryAllByLabelText(
 ) {
   checkContainerType(container)
 
+  const matcher = exact ? matches : fuzzyMatches
   const matchNormalizer = makeNormalizer({collapseWhitespace, trim, normalizer})
-  const labels = queryAllLabelsByText(container, text, {
-    exact,
-    normalizer: matchNormalizer,
-  })
-  const labelledElements = labels
-    .reduce((matchedElements, label) => {
-      const elementsForLabel = []
-      if (label.control) {
-        elementsForLabel.push(label.control)
-      }
-      /* istanbul ignore if */
-      if (label.getAttribute('for')) {
-        // we're using this notation because with the # selector we would have to escape special characters e.g. user.name
-        // see https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector#Escaping_special_characters
-        // <label for="someId">text</label><input id="someId" />
-
-        // .control support has landed in jsdom (https://github.com/jsdom/jsdom/issues/2175)
-        elementsForLabel.push(
-          container.querySelector(`[id="${label.getAttribute('for')}"]`),
+
+  const matchingLabelledElements = Array.from(container.querySelectorAll('*'))
+    .filter(element => element.hasAttribute('aria-labelledby'))
```

Comment:
I think we might be able to clean up some stuff (and reduce the number of loops) by changing this filter to include elements with the `.label` property:

```suggestion
    .filter(element => element.label || element.hasAttribute('aria-labelledby'))
```

This would require the `reducer` function handle the `.label` elements a little differently, but I think it would allow us to remove some other code as well (specifically all the code for `matchingElementsByLabels`).

Reply:
I am going to try what you suggested as soon as possible.

Reply:
Hey @kentcdodds !
For sure I am missing something but so far I have not been able to make the change you suggested me. 
Trying to include the elements with the `.label` property in the filter you highlighted it is not working because the element labeled by a `label` element put around them have the `.label` property undefined.
I have tried to show it in [this](https://codesandbox.io/s/nostalgic-benz-fnxsm?file=/src/__tests__/index.js) sandbox.

Again, I am aware there is something I am not doing in the right way but I am not able to get it.

Reply:
Oh, sorry, that's a typo. It should be `.labels`. This is a `NodeList` of all the labels. If there are multiple, then it looks like the Chrome Accessibility DevTools combines them:

![image](https://user-images.githubusercontent.com/1500684/87700844-3b0b5480-c754-11ea-80b0-9b4a2a5df752.png)

https://jsbin.com/pudalunaxu/1/edit?html,output

```html
<div>
  <label for="my-input">One</label>
  <label for="my-input">Two</label>
  <input id="my-input" />
</div>
```

So I think that's probably what we should do as well. Combine the text for all labels in the NodeList, then compare that against the given TextMatch.

Does that make sense?

Reply:
Yes, it makes a lot of sense.
I have just pushed some code using `.labels`.

I am not so sure I have done all the work because I am not obtaining all the labels in the NodeList through a permutation but just join their values (and join values in the NodeList removing an item a time). I don't know if you meant this with `combine`.